supertokens_python/recipe/multifactorauth/index.py

```diff
@@ -0,0 +1,146 @@
+from typing import Any, Dict, List, TypedDict
+
+from supertokens_python.recipe.multiFactorAuth.asyncio import MultiFactorAuthRecipe
+from supertokens_python.recipe.multitenancy.asyncio import (
+    get_tenant_id,
+    get_tenant_id_from_user_context,
+    get_user,
+    get_user_context,
+)
+from supertokens_python.recipe.multitenancy.exceptions import (
+    SuperTokensUnknownUserIdError,
+)
+from supertokens_python.recipe.multitenancy.recipe import MultitenancyRecipe
+from supertokens_python.recipe.multitenancy.utils import get_user_context
+from supertokens_python.recipe.session import SessionContainer
+from supertokens_python.recipe.session.asyncio import SessionContainer
+from supertokens_python.recipe.session.interfaces import SessionContainer
+from supertokens_python.utils import get_user_context
+
+from .recipe import MultiFactorAuth, Recipe
+from .utils import update_and_get_mfa_related_info_in_session
+
+
+class FactorIds(TypedDict):
+    EMAILPASSWORD: str
+    OTP_EMAIL: str
+    OTP_PHONE: str
+    TOTP: str
+    OAUTH: str
+
+
+FACTOR_IDS = FactorIds(
+    EMAILPASSWORD="emailpassword",
+    OTP_EMAIL="otp-email",
+    OTP_PHONE="otp-phone",
+    TOTP="totp",
+    OAUTH="oauth",
+)
+
+
+async def assert_allowed_to_setup_factor_else_throw_invalid_claim_error(
+    session: SessionContainer, factor_id: str, user_context: Dict[str, Any] = None
+) -> None:
+    ctx = get_user_context(user_context)
```

Comment:
If you see the other recipes, it this is done differently.

Reply:
Done